File: 0090-subsets-ii/0090-subsets-ii.java
import java.util.*;

class Solution {
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> result = new ArrayList<>();
        generate(nums, 0, new ArrayList<>(), result);
        return result;
    }

    private void generate(int[] nums, int idx, List<Integer> subset, List<List<Integer>> result) {
        if (idx == nums.length) {
            result.add(new ArrayList<>(subset));
            return;
        }
        subset.add(nums[idx]);
        generate(nums, idx + 1, subset, result);
        subset.remove(subset.size() - 1);  
        while (idx + 1 < nums.length && nums[idx] == nums[idx + 1]) {
            idx++;
        }
        generate(nums, idx + 1, subset, result);
    }
}

File: 0090-subsets-ii/0090-subsets-ii.py
class Solution(object):
    def subsetsWithDup(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        def helper(nums,i,curr,sets):
            if(i == len(nums)):
                sets.add(tuple(curr))
                return
            curr.append(nums[i])
            helper(nums,i+1,curr,sets)
            curr.pop()
            helper(nums,i+1,curr,sets)
        nums.sort()
        a = set()
        curr = []
        helper(nums,0,curr,a)
        return list(a)